src/client/Client.cc

```diff
@@ -4814,10 +4814,19 @@ void Client::invalidate_snaprealm_and_children(SnapRealm *realm)
 SnapRealm *Client::get_snap_realm(inodeno_t r)
 {
   SnapRealm *realm = snap_realms[r];
-  if (!realm)
+  int count = 1;
+
+  if (!realm) {
     snap_realms[r] = realm = new SnapRealm(r);
-  ldout(cct, 20) << __func__ << " " << r << " " << realm << " " << realm->nref << " -> " << (realm->nref + 1) << dendl;
-  realm->nref++;
+
+    // Do not release the global snaprealm until unmounting.
+    if (r == CEPH_INO_GLOBAL_SNAPREALM) {
+      count++;
```

Comment:
If I am understanding this correctly, you could do away with the local `count` variable, isn't it?

Reply:
Hi Venky,

Sorry, I don't get this. Currently the `count` is already a local variable.

Actually I can just do the `realm->nref++` here. And then at the end do this again just like what it was before. The reason I add a new `count` variable here is only for the `ldout(ctt, 20)...` below.

Reply:
Well, you can log realm->count before and after too (for debugging I guess) and avoid using `count`.

Reply:
Do you mean something like:

```
ldout(cct, 20) << __func__ << " " << r << " " << realm << " nref was " << realm->nref << dendl;
...
ldout(cct, 20) << __func__ << " " << r << " " << realm << " nref now is " << realm->nref << dendl;
```

??

Reply:
Right. My point is, you can avoid introducing local `count` if its just for debug purpose.

I know its just a smal local var, but.....

Reply:
Sure, will update it.